### PR TITLE
Allow specify $attributeNames as a string for yii\base\Model validate()

### PR DESCRIPTION
**Pull request by developeruz**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | not sure
| Tests pass?   | yes
| Fixed issues  | #8752

### CODE REVIEW

File: framework/base/Model.php [hunk 1]
```diff
@@ -356,6 +356,10 @@ public function validate($attributeNames = null, $clearErrors = true)
             $attributeNames = $this->activeAttributes();
         }
 
+        if(!is_array($attributeNames)) {
```
Comment (MysteryDragon) [inline framework/base/Model.php]: @developeruz maybe it's better to put space between `if` and `(` for PSR-2 stuff?

File: tests/framework/base/ModelTest.php [hunk 2]
```diff
@@ -462,10 +462,21 @@ public function testValidateWriteOnly()
...
+        $model->name = 'Some value';
+        $this->assertTrue($model->validate(['name']), 'Should validate only name attribute');
+        $this->assertTrue($model->validate('name'), 'Should validate only name attribute');
+        $this->assertFalse($model->validate(), 'Should validate all attributes');
+    }
 }
 
 class ComplexModel1 extends Model
 {
+    public $name, $description, $id, $is_disabled;
```
Comment (MysteryDragon) [inline tests/framework/base/ModelTest.php]: Maybe place every property at new line for [PSR-2 compability](http://www.php-fig.org/psr/psr-2/#42-properties)?

**Review by samdark — APPROVED** [review]

File: framework/base/Model.php [hunk 3]
```diff
@@ -356,6 +356,10 @@ public function validate($attributeNames = null, $clearErrors = true)
             $attributeNames = $this->activeAttributes();
         }
 
+        if (!is_array($attributeNames)) {
+            $attributeNames = (array)$attributeNames;
+        }
+
```
Comment (rob006) [inline framework/base/Model.php]: You could remove `if` and always cast to array.

File: framework/base/Model.php [hunk 4]
```diff
@@ -329,7 +329,7 @@ public function attributeHints()
      * Errors found during the validation can be retrieved via [[getErrors()]],
      * [[getFirstErrors()]] and [[getFirstError()]].
      *
-     * @param array $attributeNames list of attribute names that should be validated.
+     * @param array|string $attributeNames attribute name or list of attribute names that should be validated.
```
Comment (rob006) [inline framework/base/Model.php]: `string[]|string` could be used.

File: framework/base/Model.php (hunk 3 above)
Comment (developeruz) [reply framework/base/Model.php]: Done, thank you, it helped to solve codeclimate issue

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**